### ISSUE TITLE
[Bug]: State not updating

### ISSUE THREAD

**Issue by ArshSoni** [bvis/aegis-hass#129]
### Describe the bug

When the Ajax alarm is armed via the Ajax or Ajax Pro app for either the entire house or for groups, the state in home assistant is not updating. 

The only way for it to update is if the integration is reloaded in settings. 

What could be causing this?

### Steps to reproduce

Alarm and view the state in home assistant.

### Expected behavior

Should update in home assistant to show as alarmed

### Home Assistant version

Latest

### Integration version

Latest

### Installation method

HACS

### App label (co-branded app)

Ajax

### FCM push notifications configured?

Yes

### Relevant logs

```text

```

**Comment by bvis** [comment]
Thanks @ArshSoni — the symptom you describe (state stale until you reload the integration, then correct) is the textbook signature of FCM push not delivering arm/disarm events to this install. The integration polls Ajax every 5 min, so reloading forces an immediate poll and you see the correct state right away, but in between two polls only the FCM push channel can update the state in near-real-time. If FCM isn't being delivered, the state appears stuck.

Three things to nail down the cause:

1. Add the following under `logger:` in `configuration.yaml`, restart HA, arm from the mobile app, wait ~30 s, then disarm. Share the snippet between those two actions:

   ```yaml
   logger:
     logs:
       custom_components.aegis_ajax: debug
       custom_components.aegis_ajax.notification: debug
   ```

   The lines I'm looking for are `Push notification received: persistent_id=...` (push arrived) and whatever follows it. If there's nothing between your arm and disarm in that log, FCM is not delivering — that's a different fix path (registration or app-label mismatch) than if the push is arriving and parsing wrong.

2. **Which app exactly?** The dropdown shows "Ajax" and "Ajax Pro" as separate options because they're different FCM projects on Ajax's side. If you picked "Ajax" in the integration but actually use "Ajax PRO Desktop" / "Ajax Pro" on your phone (or vice-versa), the push won't be delivered to this integration even though both apps see the arm.

3. Open **Settings → Devices & Services → Aegis for Ajax → System Health card** (or run **Settings → Repairs → ⋮ → System information**) and paste the "Pushes received" and "Last push" lines. That tells me whether FCM has *ever* delivered anything to this install or it's been silent since startup.

With those three I can tell whether to scope this as a registration bug, an app-label mismatch, or a parsing issue.

**Comment by bvis** [comment]
Quick cross-reference for context: @Permudious in #119 has just confirmed the exact same symptom on his install — `Push notification received` lines never appear in the log even with `notification: debug` enabled and the device event fired. So the most likely root cause for both reports is FCM not reaching this kind of install (not a per-event parsing bug). The three diagnostic items I asked for above are the same ones I asked there; whichever of you posts first will probably unblock both threads.

**Comment by bvis** [comment]
@ArshSoni — `1.3.0-beta.9` is out via HACS with a targeted change for exactly this symptom. It adds a Repair card that fires when FCM credentials are missing on an Aegis entry, and promotes the corresponding log line from INFO to WARNING so it's visible at HA's default log level.

To check whether that's been the cause:

1. Update HACS to `1.3.0-beta.9` and restart HA.
2. Open **Settings → Repairs**. If you see a card titled *"Push notifications not configured — real-time events disabled"*, FCM has never been set up for this account — that's the reason state only refreshes on reload (which forces a polling cycle), because pushes are the channel that would carry an arm/disarm in real time. The card itself is the fix flow: click it, enter the four FCM values (Project ID / App ID / API Key / Sender ID) for your cobrand, submit. The integration reloads and state should track the mobile app in seconds from then on.
3. If you don't see that Repair, check `home-assistant.log` at the default log level (no debug filter needed) for one of these three WARNING lines:
   - `FCM credentials not configured — push notifications disabled, real-time events ... will not reach HA` → same case as the Repair card.
   - `Failed to start FCM push client` / `FCM registration failed` → keys are set but wrong; the `fcm_credentials_invalid` Repair will surface that and let you fix them.
   - Neither line at all → FCM is configured and the push client started; the problem is elsewhere.

**If you see neither the Repair card nor any of the three WARNING lines** — paste a fresh log snippet from boot here. Any FCM-related state of this install should now leave a WARNING line by default; if none appears, that's a separate bug I want to see.

The README has a section on how to extract the four FCM values for your cobranded Ajax app.

**Comment by Hansontech190** [comment]
Unfortunately I'm having the same issue again with the state not updating.  The issue with sub-key 11 in issue #111 seems to have been resolved but the status is still problematic.  Logs attached for reference.  I'm on beta 8 so will try updating to beta 9.

The activity log for the hub doesn't show any arming or disarming, only a "security event" 22 hours ago.

[aegis_ajax_debug_extract_v2.log](https://github.com/user-attachments/files/27751628/aegis_ajax_debug_extract_v2.log)

**Comment by bvis** [comment]
Thanks @Hansontech190 — good news on the sub-key 11 line being gone, that confirms `beta.8` did its job for the HTS spam. The state-not-updating part is a separate channel (FCM push), and the diagnostic surface for it just landed in `1.3.0-beta.9`: a fixable Repair card + a default-level WARNING log when FCM credentials are missing.

Could you:

1. Update HACS to **`1.3.0-beta.9`** and restart HA.
2. Check **Settings → Repairs** — if a card titled *"Push notifications not configured — real-time events disabled"* appears, that alone explains the symptom and the card itself is the fix flow ([how to obtain FCM credentials](https://github.com/bvis/aegis-hass#how-to-obtain-fcm-credentials)).
3. Either way, post a fresh log snippet from boot. Without the `beta.9` logs I can't tell whether you're in the same FCM-not-configured state as #119 or whether something else is going on, and your `beta.8` log doesn't have the FCM-state lines yet.

That should narrow it down quickly.

**Comment by Hansontech190** [comment]
> 2\. Check **Settings → Repairs** — if a card titled _"Push notifications not configured — real-time events disabled"_ appears, that alone explains the symptom and the card itself is the fix flow

Yes HA showed this as a repair after updating to beta 9 and restarting.

Logs attached after reboot.

[home-assistant_aegis_ajax_2026-05-14T11-52-40.860Z.log](https://github.com/user-attachments/files/27756834/home-assistant_aegis_ajax_2026-05-14T11-52-40.860Z.log)

**Comment by bvis** [comment]
Thanks @Hansontech190 — the `beta.9` log confirms exactly what we expected: three `FCM credentials not configured — push notifications disabled` WARNINGs and zero `Push notification received` lines. So the root cause of "state not updating" on your install is identified: real-time events were never reaching HA because FCM was off, and HA was relying on the 5-minute polling cycle.

The next step (entering credentials in the Repair card) is being tracked in #131, so I'll close this one to avoid two parallel threads on the same install. Continuing the conversation there.
